actors/src/main/kotlin/ktx/actors/events.kt

```diff
@@ -88,6 +89,20 @@ inline fun <T : Actor> T.onClickEvent(
   return clickListener
 }
 
+/**
+ * Attaches a [ClickListener] to this tree.
+ * @param lsitener invoked each time a node in this tree is clicked. Consumes the triggered [InputEvent] and the
+ * [Tree] the listener was originally attached to as `this`. The received Node is the node that was clicked.
+ * @return [ClickListener] instance.
+ */
+inline fun <T : Tree<*, *>> T.onItemClick(crossinline listener: ((Tree.Node<*, *, *>) -> Unit)): ClickListener {
```

Comment:
You might be able to add more generic types so that the listener receives a specific type of node, but I'm not sure how useful that would be in this case. Depends on whether the tree is constructed via our DSL or manually with a custom node class.

Reply:
You can try to introduce more generic type, so that there's no need to use wildcard types for node, but if it proves cumbersome to use, then feel free to scrap it.